src/app/@ansyn/ol/plugins/annotations/annotations-context-menu/models/free-text-attribute.ts

```diff
@@ -1,19 +1,15 @@
 import { ControlType } from './control-type.enum';
 import { AttributeBase } from './attribute-base';
-import { IKeyValuePair } from './key-value.interface';
+import { IAttributeData } from './attribute-data.interface';
 
 export class FreeTextAttribute extends AttributeBase<string> {
-	type = ControlType.FreeText;
-	options?: IKeyValuePair<string>[];
 	constructor(
-		options: {
-			key?: string;
-			label?: string;
-			type?: ControlType;
-			value?: string;
-		} = {}
+		data: IAttributeData
 	) {
-		super(options);
-		this.options = options['options'] || [];
+		super(data);
+		this.type = ControlType.FreeText;
+		if (!!data.value && typeof data.value !== 'string') {
```

Comment:
try to use https://stackoverflow.com/questions/1303646/check-whether-variable-is-number-or-string-in-javascript

Reply:
in the link you sent they are using typeof keyword same as here